src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,37 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create({ ...data, username: data.user });
+    await msg.populate('user', 'username');
+    console.log('Chat message saved:', msg);
+
+    // Return the _id of the saved message
+    const savedMessage = {
+      ...data,
+      _id: msg._id, // Include _id in the message
+      username: data.username,
+    };
```

Comment:
I think instead of creating a variable that combines the incoming `data` and the username from `msg.user.username`, we can instead emit `msg` as the payload on line 31 directly. That should have all the information that the frontend needs to render the messages.

Reply:
<img width="612" alt="name" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/dde80ceb-e550-4afc-a115-8c6d50f65906"> 
Displays in terminal 'name'.

Reply:
> <img alt="name" width="612" src="https://user-images.githubusercontent.com/95896159/274768499-dde80ceb-e550-4afc-a115-8c6d50f65906.png"> 

Displays in terminal 'name' only, the 'username' does not work.

Reply:
Yes because on line 11 you've populated only `'name'` . If you want username to be populated onto the `ChatMessage.user` object from the `User` model, that line should be `await msg.populate('user', 'username')` which I think you did have there before :thinking: 

Or if you want both of those fields you can pass in an array: `await msg.populate('user', ['name', 'username'])`. Or if you really really need the _full_ user object, then you can omit the second argument: `await msg.populate('user')`. But generally it's best (especially over websocket which we want to keep speedy) to just send the fields that the client needs.

Reply:
Ah I see in the frontend you're using `user.name` now so I think you intentionally changed this to 'name' since the last review; all good then as long as frontend and backend match in what they're expecting :+1: https://github.com/Code-the-Dream-School/dd-prac-team4-front/pull/52/files#diff-dc044c6ce14501135532d751cb2c59b15079b67646f1634abe6dd9c739c899b4R125

Reply:
<img width="658" alt="name" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/6afdeb80-682c-43ee-8aca-a24a6980e1a8">
<img width="633" alt="username" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/a3e973ca-4235-414c-96be-70968ee0cecf">

Reply:
> I think instead of creating a variable that combines the incoming `data` and the username from `msg.user.username`, we can instead emit `msg` as the payload on line 31 directly. That should have all the information that the frontend needs to render the messages.
I probably didn’t understand correctly earlier, I thought I needed to make these changes.

Reply:
![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/d6783333-5da4-44fd-bb3d-f3522ab62c1f)

It seems that particular user does not have a username in the database; so it makes sense that nothing was returned when that field was populated. We added `username` as a required field somewhere in the middle of the project, so users that were created before that may not have a user name unless we go back and fix them. :upside\_down\_face:  Sorry that caused some confusion!